.github/workflows/simple_per_test.yml

```diff
@@ -0,0 +1,366 @@
+          echo "::set-env name=MAX32655_STEP::${MAX32655_STEP}"
+          echo "::set-env name=MAX32665_STEP::${MAX32665_STEP}"
+          echo "::set-env name=MAX32690_STEP::${MAX32690_STEP}"
+          echo "::set-env name=MAX32690_WLP_STEP::${MAX32690_WLP_STEP}"
+          echo MAX32655_STEP: ${MAX32655_STEP}
+          echo MAX32665_STEP: ${MAX32665_STEP}
+          echo MAX32690_STEP: ${MAX32690_STEP}
+          echo MAX32690_WLP_STEP: ${MAX32690_WLP_STEP}
+          echo
+
+      # Runs a set of commands using the runners shell
+      - name: max32655_simple_per_test
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: "true"
+
+        run: |
+
+          #------------------------------------------------------------------------------------------------------------
+          # Prepare the arguments for script.                    
+          MSDK=${GITHUB_WORKSPACE}
+          CHIP_UC=MAX32655
+
+          $MSDK/msdk_dev_tools/scripts/msdk_per_skip_check.sh \
+            $NO_SKIP \
+            $MSDK    \
+            $CHIP_UC
+
+          if [[ $? -ne 0 ]]; then
+            echo "SKIPPED."
+            exit 0
+          fi
+
+          #------------------------------------------------
+          # Prepare the arguments for function call
+
+          if [ "${DO_MAX32655}" == "0" ]; then
+            echo "Skip the test for MAX32655."
+            exit 0
+          fi
+
+          BRD2_CHIP_LC=max32655
+
+          MSDK=${GITHUB_WORKSPACE}
+          BRD1=nRF52840_1
+          BRD2=max32655_board_2
+
+          CURR_TIME=$(date +%Y-%m-%d_%H-%M-%S)
+
+          CURR_JOB_FILE=/home/$USER/Workspace/Resource_Share/History/msdk_simple_per_test_${CURR_TIME}_${BRD2_CHIP_LC}.txt
+          echo "::set-env name=CURR_JOB_FILE::${CURR_JOB_FILE}"
+
+          CURR_LOG=/home/$USER/Workspace/Resource_Share/Logs/msdk_simple_per_test_${CURR_TIME}_${BRD2_CHIP_LC}.log          
+          echo "::set-env name=CURR_LOG::${CURR_LOG}"
+
+          RESULT_PATH=~/Workspace/ci_results/per
+          res=${RESULT_PATH}/msdk-${CURR_TIME}
+          all_in_one=${res}_${BRD2_CHIP_LC}.csv
+          echo "::set-env name=all_in_one::${all_in_one}"
+
+          #------------------------------------------------
+          ${MSDK}/msdk_dev_tools/scripts/board_per_test.sh \
+            $MSDK      \
+            $BRD1      \
+            $BRD2      \
+            $CURR_TIME \
+            $(realpath ${CURR_JOB_FILE}) \
+            $(realpath ${CURR_LOG})      \
+            $(realpath ${all_in_one})    \
+            "${MAX32655_PKG_RA}"         \
+            "${MAX32655_PHY_RA}"         \
+            ${MAX32655_STEP}             \
+            2>&1 | tee -a ${CURR_LOG}
+          
+          if [[ $? -ne 0 ]]; then
+            exit 1
+          fi
+          #------------------------------------------------          
+
+      # Unlock even when cancelled or failed
+      - name: unlock_max32655
```

Comment:
Where are we doing the locking? We need to make sure that we don't unlock if the lock was unsuccessful.

Reply:
I put all the same operations into a script file so that if there's any change, we only need to change at one place instead of changes everywhere.

Reply:
The locking/unlocking are in the script file. It can handle the case if the test failed or was canceled.